packages/next/next-server/lib/router/router.ts

```diff
@@ -785,12 +785,7 @@ export default class Router implements BaseRouter {
           this._resolveHref(parsedHref, pages)
 
           if (pages.includes(parsedHref.pathname)) {
-            return this.change(
-              'replaceState',
-              destination,
-              destination,
-              options
-            )
+            return this.change(method, destination, destination, options)
```

Comment:
In previous code, this replaced history of **the origin page** not the one with redirect settings with new page. So, I think it is better to use "pushState" and  use "replaceState" in the only case of first navigation of SSG.